libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -1218,23 +1373,29 @@ ManagedQuery::_extend_and_evolve_schema_with_details(
     ArrowSchema* value_schema,
     ArrowArray* value_array,
     std::string column_name,
+    bool deduplicate,
     Enumeration enmr,
     ArraySchemaEvolution& se) {
-    // We need to check if we are writing any new enumeration values. If so,
-    // extend and evolve the schema. If not, just set the write buffers to the
-    // dictionary's indexes as-is
+    if (value_array->n_buffers != 2) {
```

Comment:
this also raises the question of how you handle the nullable buffer - what happens if the user sends a value array with positions nulled?  Is that allowed?

I think the simplest thing is disallow it and perform the test at the very top-level (_extend_enumeration)

Reply:
pushed